cpp/src/dictionary/dictionary_factories.cu

```diff
@@ -89,4 +91,71 @@ std::unique_ptr<column> make_dictionary_column(std::unique_ptr<column> keys_colu
                                   std::move(children));
 }
 
+namespace {
+
+/**
+ * @brief This functor maps signed type_ids to unsigned counterparts.
+ */
+struct signed_to_unsigned_type_fn {
```

Comment:
std C++ has `std::make_unsigned<T>`. Since I like to be consistent with standards where possible, maybe this should be called `make_unsigned_fn`...